include/private/firebird/common.h

```diff
@@ -98,14 +98,15 @@ double round_for_isc(double value)
 }
 
 template<typename T1>
-void to_isc(void * val, XSQLVAR * var, int x_scale = 0)
+void to_isc(void * val, XSQLVAR * var, short x_scale = 0)
 {
     T1 value = *reinterpret_cast<T1*>(val);
     short scale = var->sqlscale + x_scale;
     short type = var->sqltype & ~1;
     long long divisor = 1, multiplier = 1;
 
-    if ((std::numeric_limits<T1>::is_integer == false) && scale >= 0 &&
+    static bool is_integer = std::numeric_limits<T1>::is_integer;
```

Comment:
P.S. Although to be honest, I don't think it makes sense to really factor it out, I'm almost certain that the compiler inlines this in the optimized build anyhow.